src/Angular.js

```diff
@@ -765,23 +767,37 @@ ES6MapShim.prototype = {
     }
     return (this._lastIndex = (this._keys.indexOf(this._lastKey = key)));
   },
+  _transformKey: function(key) {
+    return isNumberNaN(key) ? nanPlaceholder : key;
+  },
   get: function(key) {
-    var idx = this._idx(key);
+    var idx = this._idx(this._transformKey(key));
     if (idx !== -1) {
       return this._values[idx];
     }
   },
   set: function(key, value) {
+    key = this._transformKey(key);
     var idx = this._idx(key);
     if (idx === -1) {
       idx = this._lastIndex = this._keys.length;
-      this._lastKey = key;
```

Comment:
It might not be obvious if you are trying to figure out why it being deleted, but if you look at the code (without knowing it was ever there), I think it is clearer without the comment.
I am fine adding if you still think it helps, though.